context-free.md

```diff
@@ -1,22 +1,37 @@
 # Context Free
 
 ##  Who is this programming language for?
-
+This programming language is for people who want to create digital art.
 
 ## What is easy to do in this language? Why is it easy?
-
+Recursively creating shapes is easy because shapes are created inside rules.
 
 ## What is hard to do in this language? Why is it hard?
-
+It was very hard for me to learn it, taking well over an hour.  It requires
+memorization of syntax and avaiable rules unlike a simple program such as paint
+where you can see what is available for use.  Additional memorization is
+required to know what transforamtions you have available for your shape (such
+as using hue to color things is less obvious than something like color). Drawing
+like one would with a pencil is quite hard because everything has to be
+programmed in and cannot be drawn using mouse.
```

Comment:
This is an interesting tradeoff - there are some programs that take the opposite route (draw something and then generate the code for you) and are especially common with certain GUI toolkits. The downside is that often those code generation tools generate code that is hard to read, maintain, and modify. 

I think this says a lot about the target audience - the programs I just mentioned are aimed at a designer, not a developer, while this language is aimed at developers first.

I think a great extension or tool for this language, however, would be a real-time (or as close as possible) interpreter that drew as you coded (or coded as you drew).